### DISCUSSION TITLE
Building with "feature/vitis_hls_2021.2" branch

### DISCUSSION THREAD

**Discussion by bestamigunay** [Xilinx/finn#521] [Q&A] [answered]
When I try to build with Vivado 2021.2 & Vitis HLS 2021.2 and  `feature/vitis_hls_2021.2` branch, I get this error:

```
Running step: step_qonnx_to_finn [1/17]
Running step: step_tidy_up [2/17]
Running step: step_streamline [3/17]
Running step: step_convert_to_hls [4/17]
Running step: step_create_dataflow_partition [5/17]
Running step: step_target_fps_parallelization [6/17]
Running step: step_apply_folding_config [7/17]
Running step: step_generate_estimate_reports [8/17]
Running step: step_hls_codegen [9/17]
Running step: step_hls_ipgen [10/17]

multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/opt/conda/lib/python3.8/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/opt/conda/lib/python3.8/multiprocessing/pool.py", line 48, in mapstar
    return list(map(*args))
  File "/workspace/finn/src/finn/transformation/fpgadataflow/hlssynth_ip.py", line 69, in applyNodeLocal
    inst.ipgen_singlenode_code()
  File "/workspace/finn/src/finn/custom_op/fpgadataflow/hlscustomop.py", line 360, in ipgen_singlenode_code
    assert os.path.isdir(
AssertionError: IPGen failed: /tmp/finn_dev_best/code_gen_ipgen_FMPadding_Batch_7_veme0gjz/project_FMPadding_Batch_7/sol1/impl/ip not found. Check log under /tmp/finn_dev_best/code_gen_ipgen_FMPadding_Batch_7_veme0gjz
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/workspace/finn/src/finn/builder/build_dataflow.py", line 166, in build_dataflow_cfg
    model = transform_step(model, cfg)
  File "/workspace/finn/src/finn/builder/build_dataflow_steps.py", line 426, in step_hls_ipgen
    model = model.transform(HLSSynthIP())
  File "/workspace/finn-base/src/finn/core/modelwrapper.py", line 141, in transform
    (transformed_model, model_was_changed) = transformation.apply(
  File "/workspace/finn-base/src/finn/transformation/base.py", line 105, in apply
    new_nodes_and_bool = p.map(self.applyNodeLocal, old_nodes, chunksize=1)
  File "/opt/conda/lib/python3.8/multiprocessing/pool.py", line 364, in map
    return self._map_async(func, iterable, mapstar, chunksize).get()
  File "/opt/conda/lib/python3.8/multiprocessing/pool.py", line 771, in get
    raise self._value
AssertionError: IPGen failed: /tmp/finn_dev_best/code_gen_ipgen_FMPadding_Batch_7_veme0gjz/project_FMPadding_Batch_7/sol1/impl/ip not found. Check log under /tmp/finn_dev_best/code_gen_ipgen_FMPadding_Batch_7_veme0gjz

> /opt/conda/lib/python3.8/multiprocessing/pool.py(771)get()
    769             return self._value
    770         else:
--> 771             raise self._value
    772 
    773     def _set(self, i, obj):


```

And when I check the log under `/tmp/finn_dev_best/code_gen_ipgen_FMPadding_Batch_7_veme0gjz`, I see this error:

`ERROR: [HLS 207-5492] RAM_S2P + URAM is not valid value for option BIND_STORAGE's option 'type + impl' (/workspace/finn-hlslib/slidingwindow.h:120:49)`

Could anyone help me about this problem, please? Thank you!

**Comment by bestamigunay** [comment]
Also, it works well with Vivado & Vitis HLS 2020.1 and `main` branch. However, I need to use `2021.1` for Kria KV260.

**Comment by maltanar** [comment, marked as answer]
The `feature/vitis_hls_2021.2` branch still has numerous Vitis HLS issues we're working through. We've seen this particular problem when trying to synthesize for devices that don't have any URAM, though this shouldn't be an issue for the Kria KV260. A temporary workaround is to change the storage specifier pragmas in `finn-hlslib/slidingwindow.h` to use the "old-style" Vivado HLS pragmas, but a better solution will be coming with Vitis HLS 2022.1 and the new FINN release.

**Reply by bestamigunay** [reply]
Thank you for your answer Yaman! Actually, I did some wrong modifications for the definition of the KV260, but I solved it and synthesized succesfully. 
However, I still wonder that how should I set the AXI HP port width in `util/basic.py` ? Should it be 128 for the KV260? I set it 128, but I am not sure about it. Actually, I am asking this question, because my CNN works very well on Pynq-Z2 board (synthesized with HLS 2020.1) but behaves different on KV260 board (synthesized with HLS 2021.2). I think the problem is either the AXI HP port width or the HLS version. Do you have any comment about it? Thank you!
